actors improvements

abom opened the pull request:
also fix gedis code parsing

GlenDC commented on Jumpscale/clients/blockchain/tfchain/TFChainCapacity.py:
```diff
@@ -128,10 +129,40 @@ def reserve_zos_vm(self, email, threebot_id, location, size=1, source=None, refu
+        _validate_reservation_vm(reservation)
+        return self._process_reservation(reservation, threebot_id, source=source, refund=refund)
+
+    def reserve_zdb_namespace(self, email, threebot_id, location, size=1,
```
Misses a docstring.

GlenDC replied:
It is not clear to me what location is supposed to be.

zaibon replied:
Same as the location for a 0-OS VM, can be a node id or a farm name.
I added doc sting so it's more clear

GlenDC commented on Jumpscale/clients/blockchain/tfchain/TFChainCapacity.py:
```diff
@@ -128,10 +129,40 @@ def reserve_zos_vm(self, email, threebot_id, location, size=1, source=None, refu
+
+    def reserve_reverse_proxy(self, email, threebot_id, domain, backend_urls,
+                              source=None, refund=None):
+        reservation = j.data.schema.get(url='tfchain.reservation.reverse_proxy').new(data={
```
Misses a docstring.

GlenDC commented on Jumpscale/clients/blockchain/tfchain/TFChainCapacity.py:
```diff
@@ -157,16 +188,48 @@ def _process_reservation(self, reservation, threebot_id, source=None, refund=Non
+        if not getattr(reservation, field):
+            raise ValueError("field '%s' cannot be empty" % field)
+
+    if not reservation.mode or reservation.mode not in ['seq', 'user', 'direct']:
```
Do we need to validate it manually like this? Don't schemas allow to validate this automatically? Seems a bit silly otherwise (not to say error prone).

xmonader committed:
clean off the html in the food_get flows and add example for food_get_form

xmonader committed:
Add j.data.countries

xmonader committed:
add drop_down_country

ranatrk committed:
Fix loading in food get chatflow

abom committed:
fix using actor_method with method without docstrings...

also fix gedis code parsing

ranatrk committed:
Add actor method decorator to actor methods in threebot base actors

waleedhammam committed:
add autocomplete_drop_down to chatbot

serboctor committed:
Override setattr for jsxobject to make sure only valid properties are set

serboctor committed:
Merge branch 'development' into development_bcdbautoupdate

abom committed:
add actor_method to ebcdb

abom committed:
fix package_manager add when an error is raised

abom committed:
Merge branch 'development' into development_actorsimprovement

serboctor committed:
Add path url to location proxy

abom committed:
log the error when adding package

ranatrk committed:
Change log in package manager

Commit:
show de debug

Commit:
Merge branch 'development_bcdbautoupdate' into development

* development_bcdbautoupdate:
  Add path url to location proxy
  Override setattr for jsxobject to make sure only valid properties are set

Commit:
Merge branch 'development' into development_actorsimprovement

* development:
  show de debug
  wireguard tool: add method to generate wg keys to use for 0-OS networking
  Add path url to location proxy
  Override setattr for jsxobject to make sure only valid properties are set

Commit:
merge conflict resolve

Commit:
Merge branch 'development_osx' into development_actorsimprovement

* development_osx:
  merge conflict resolve

# Conflicts:
#	JumpscaleCore/clients/tfgrid_registry/TFGridRegistryClient.py

Commit:
Merge branch 'development_chatbot' into development_actorsimprovement

* development_chatbot:
  add autocomplete_drop_down to chatbot
  Fix loading in food get chatflow
  add drop_down_country
  Add j.data.countries
  clean off the html in the food_get flows and add example for food_get_form

Commit:
changes for better export/import

Commit:
bcdb fixes

Commit:
bcdb fixes

xmonader committed:
fixes for import/export

Commit:
refactor loading package

Commit:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

* 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core:
  fixes for import/export

serboctor committed:
Undo wrong merge

Commit:
Adding tests to registry client

Commit:
Adding tests to registry client 2

serboctor committed:
Fix import data bug

serboctor committed:
Fix import for subobjects

serboctor committed:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

Commit:
threebotserver security fixes & refactoring

Commit:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

* 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core:
  Fix import for subobjects
  Fix import data bug
  Adding tests to registry client 2
  Adding tests to registry client
  Undo wrong merge

Commit:
some bugs fixed in loading of packages

Commit:
threebot package

Commit:
packages working now

RafyAmgadBenjamin committed:
Fix starting of threebot server

Commit:
bcdb load changes

RafyAmgadBenjamin committed:
Fixing an issue related to the names

Commit:
basic bcdb

Commit:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

* 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core:
  Fixing an issue related to the names
  Fix starting of threebot server

waleedhammam committed:
add tcprouter to threfold image

serboctor committed:
Use a fixed dummy model in bcdb import

waleedhammam committed:
add tcprouter client

waleedhammam committed:
update tcprouter client readme

Commit:
fixes threebotserver

Commit:
Merge branches 'development_actorsimprovement' and 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

* 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core:
  update tcprouter client readme
  add tcprouter client
  add tcprouter to threfold image

* 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core:
  update tcprouter client readme
  add tcprouter client
  add tcprouter to threfold image

serboctor committed:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

andrewayoub committed:
select the correct namespace after getting redis client in zdbclient

andrewayoub committed:
use any bcdb instance provided in threebot package

xmonader committed:
update wiki package to not be ignored

abom committed:
threebot.start: reset myjobs after starting zdb

Commit:
bcdb checks on empty indexed fields

RafyAmgadBenjamin committed:
Fixing an issue related to BuilderBaseClass

abom committed:
remove threebot_server_name usage from j.tools.threebot_packages

abom committed:
move indexed fields check to save()

RafyAmgadBenjamin committed:
Fixing prompt_toolkit version in InstallTools

RafyAmgadBenjamin committed:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

waleedhammam committed:
fix git log changes

andrewayoub committed:
add zdb client docs #179

abom committed:
fix wiki package loading with myjobs

abom committed:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

AhmedHanafy725 committed:
Add tests for redis config client

abom committed:
use the correct auth password for bcdb sonic client

abom committed:
fix package_manager actor

Dina-Abd-Elrahman committed:
test add files

Dina-Abd-Elrahman committed:
git client testcases

waleedhammam committed:
fix event and wait in scheduler

andrewayoub committed:
inhance packages loading and load crud actors while loading the package

andrewayoub committed:
formating

waleedhammam committed:
filling in 3git tool

andrewayoub committed:
update default packages

abom committed:
add packages with the new name convention

AhmedHanafy725 committed:
Fix redis config test expcetions

ranatrk committed:
Fix name if there is threebot in toml file for package add

Commit:
specs for registry & fixes for packages

Commit:
specs for registry & fixes for packages

Commit:
docs for jsx configure

Commit:
threebotserver start

Commit:
cannot go in background

Commit:
bcdb destroy is better now

Commit:
packages being loaded in namespace

abom committed:
fix actor method decorator calls with default arguments...

if a function is called without any arguments while having one or more
keyword arguments with default values, it would fail.

abom committed:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

abom committed:
re-add getfullname to package_manager.add method

abom committed:
Update auto-generated locations of threebot package to follow the new naming conventions

abom committed:
remove package manager actor (moved to package)

Commit:
model add with schema's

Commit:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

* 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core:
  remove package manager actor (moved to package)
  Update auto-generated locations of threebot package to follow the new naming conventions
  re-add getfullname to package_manager.add method
  fix actor method decorator calls with default arguments...

Commit:
mods for schema manipulation for packages, not doing anything yet but prepared

andrewayoub committed:
move base actors and chatflows to bas package

rkhamis committed:
Revert "model add with schema's"

This reverts commit 10d73112e9dbb31e3319bf63a25f05911613f499.

abom committed:
ingore node_modules when loading packages

waleedhammam committed:
add process to threegit tool

Hamdy committed:
[myjobs improvements]

Dina-Abd-Elrahman committed:
Sonic client

reformatted JumpscaleCore/clients/tests/test_sonic_client.py

Commit:
threebot

Commit:
Merge branch 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core into development_actorsimprovement

* 'development_actorsimprovement' of github.com:threefoldtech/jumpscaleX_core:
  [myjobs improvements]
